custom/icds_reports/reports/service_delivery_dashboard_data.py

```diff
@@ -0,0 +1,352 @@
+            base_dict['annaprasan_diwas_count'] = get_value_or_data_not_entered(row_data,
+                                                                                'annaprasan_diwas_count')
+            base_dict['suposhan_diwas_count'] = get_value_or_data_not_entered(row_data,
+                                                                              'suposhan_diwas_count')
+            base_dict['coming_of_age_count'] = get_value_or_data_not_entered(row_data,
+                                                                             'coming_of_age_count')
+            base_dict['public_health_message_count'] =\
+                get_value_or_data_not_entered(row_data, 'public_health_message_count')
+        elif step == 'sn':
+            base_dict = get_sn_percents(base_dict, row_data)
+            base_dict['lunch_0_days'] = get_value_or_data_not_entered(row_data, 'lunch_0_days')
+            base_dict['lunch_1_7_days'] = get_value_or_data_not_entered(row_data, 'lunch_1_7_days')
+            base_dict['lunch_8_14_days'] = get_value_or_data_not_entered(row_data, 'lunch_8_14_days')
+            base_dict['lunch_15_20_days'] = get_value_or_data_not_entered(row_data, 'lunch_15_20_days')
+            base_dict['lunch_21_24_days'] = get_value_or_data_not_entered(row_data, 'lunch_21_24_days')
+            base_dict['lunch_25_days'] = get_value_or_data_not_entered(row_data, 'lunch_25_days')
+            base_dict['pse_eligible'] = get_value_or_data_not_entered(row_data, 'pse_eligible')
+        elif step == 'pse':
+            base_dict = get_pse_percents(base_dict, row_data)
+            base_dict['pse_0_days'] = get_value_or_data_not_entered(row_data, 'pse_0_days')
+            base_dict['pse_1_7_days'] = get_value_or_data_not_entered(row_data, 'pse_1_7_days')
+            base_dict['pse_8_14_days'] = get_value_or_data_not_entered(row_data, 'pse_8_14_days')
+            base_dict['pse_15_20_days'] = get_value_or_data_not_entered(row_data, 'pse_15_20_days')
+            base_dict['pse_21_24_days'] = get_value_or_data_not_entered(row_data, 'pse_21_24_days')
+            base_dict['pse_25_days'] = get_value_or_data_not_entered(row_data, 'pse_25_days')
+            base_dict['pse_eligible'] = get_value_or_data_not_entered(row_data, 'pse_eligible')
+        return base_dict
+
+    all_row = dict()
+
+    data_length = len(data)
+    for index, row in enumerate(data):
+        base_row = base_data(row)
+        if not all_row.keys():
+            all_row = copy.deepcopy(base_row)
+        else:
+            all_row = update_all_row(all_row, base_row)
+        config['data'].append(base_data(row))
+
+    # setting location params to all
+    for location in location_fields:
+        all_row[location] = 'All'
+    # Calculating percentages for all row
+    if step == 'thr':
+        all_row = get_thr_percents(all_row, all_row)
+    elif step == 'sn':
+        all_row = get_sn_percents(all_row, all_row)
```

Comment:
why passing same argument twice?

Reply:
we are using it for calculating individual row percents too. There the dicts are different, here it is the same